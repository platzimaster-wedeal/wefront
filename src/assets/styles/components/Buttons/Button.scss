@import '../../partials/colors.scss';
@import '../../partials/fonts.scss';
@import '../../partials/shadows.scss';
@import '../../FontSize.scss';
@import '../../partials/sizes.scss';
@import '../../Breakpoints.scss';

.button {
    height: 42px;
    width: 150px;
    border-radius: 30px;
    cursor: pointer;
    box-shadow: $shadow-extra-small;
    @include font-size($font-size-normal);

    @include media(tablet) {
        height: 42px;
        width: 180px;
    }
}

.button__primary {
    background-color: $color-primary;
    color: $color-white;
    transition: .3s ease-in-out;
    box-shadow: $shadow-normal-green;
    &:hover {
        background-color: $color-primary-60;
        box-shadow: $shadow-large-green;
    }
}
.button__secondary {
    border: 1px solid $color-primary;
    background-color: $color-background;
    color: $color-black-60;
    transition: .3s ease-in-out;
    &:hover {
        background-color: $color-primary-30;
        color: $color-black-30;
        box-shadow: $shadow-normal-green;
    }
}