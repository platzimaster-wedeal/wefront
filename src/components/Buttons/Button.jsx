import React from "react";
import "../../assets/styles/components/Buttons/Button.scss";

const Button = ({ children, active = false, onClick }) => {
 return active ? (
  <button className="button button__primary" onClick={onClick}>
   {children}
  </button>
 ) : (
  <button className="button button__secondary" onClick={onClick}>
   {children}
  </button>
 );
};

export default Button;
