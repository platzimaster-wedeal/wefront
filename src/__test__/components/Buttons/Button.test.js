import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import Button from "../../../components/Buttons/Button";

describe("<Button/>", () => {
 const button = mount(<Button>Make Deal!</Button>);
 test("Render del componente Button", () => {
  expect(button.length).toEqual(1);
 });
 test("Render de children de un componente", () => {
  expect(button.find(".button").text()).toEqual("Make Deal!");
 });
});

describe("Button Snapshot", () => {
 test("Comprobar la UI del componente Button", () => {
  const button = create(<Button />);
  expect(button.toJSON()).toMatchSnapshot();
 });
});
