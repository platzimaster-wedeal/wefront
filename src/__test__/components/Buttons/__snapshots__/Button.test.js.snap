// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonAction/> ButtonAction snapshot 1`] = `
<button
  className="button-action undefined"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
    />
  </svg>
</button>
`;

exports[`Button Snapshot Comprobar la UI del componente Button 1`] = `
<button
  className="button button__secondary undefined"
  type="button"
/>
`;
